gax/src/main/java/com/google/api/gax/retrying/DirectRetryingExecutor.java

```diff
@@ -70,15 +72,30 @@ public DirectRetryingExecutor(RetryAlgorithm<ResponseT> retryAlgorithm) {
    */
   @Override
   public RetryingFuture<ResponseT> createFuture(Callable<ResponseT> callable) {
-    return new BasicRetryingFuture<>(callable, retryAlgorithm);
+    return createFuture(callable, RetryingContext.createDefault());
+  }
+
+  /**
+   * Creates a {@link RetryingFuture}, which is a facade, returned to the client code to wait for
+   * any retriable operation to complete. The future is bounded to {@code this} executor instance.
+   *
+   * @param callable the actual callable, which should be executed in a retriable context
+   * @return retrying future facade
+   */
+  @BetaApi("The surface for passing per operation state is not yet stable")
+  @Override
+  public RetryingFuture<ResponseT> createFuture(
+      @Nonnull Callable<ResponseT> callable, @Nonnull RetryingContext retryingContext) {
```

Comment:
We do not use `@Nonnull` annotation in other places. Please strongly consider removing (otherwise it rises a quesiton what is so special about this method compared to others, or even worse: that the other methods always accept null, because they are not annotated with `@Nonnull`)

Reply:
I don’t feel strongly about this, but I’ve been gradually introducing these when I update code (ie. streaming work I did before). Would you be ok keeping them if I decorated the rest of the class?

Reply:
Please let me know which way I should go

Reply:
Please remove them (the `@Nonnulls`). Or add everywhere else (but I think the choice is easy here...)

Reply:
done